xcodeproj/internal/xcodeprojinfo.bzl

```diff
@@ -298,10 +298,19 @@ https://github.com/MobileNativeFoundation/rules_xcodeproj/issues/new?template=bu
                 type = target_skip_type,
             ))
             return target.label
+
+        # `_APPLE_INTERNAL_TEST_BUNDLE_SUFFIX` should always be present here so remove it
+        label_name = info.xcode_target.label.name.replace(_APPLE_INTERNAL_TEST_BUNDLE_SUFFIX, "")
+
+        # As of https://github.com/bazelbuild/rules_apple/pull/1948 `bundle_name` can be used to
+        # name the bundle instead of the target name. That attribute is collected in `xcode_target.product.name` so
+        # in order to get a valid label here put the target name back using `generator_name` since at this
+        # point the original `test_suite` was already skipped
+        label_name = label_name.replace(info.xcode_target.product.name, ctx.rule.attr.generator_name)
```

Comment:
After the suffix is removed, isn't this just `info.xcode_target.product.name` or `ctx.rule.attr.generator_name`? So after all of this, don't we just want `ctx.rule.attr.generator_name` (and don't need to do any `replace`s at all)?

Reply:
Oh you're right, I was def over complicating this 🤔. At the end of the day `generator_name` always contains the value we need regardless of the value of the internal suffix `.__internal__.__test_bundle` and regardless of `bundle_name` being set or not so we can simply use that I guess!

So If that's the case I think we can also remove the warning above since if we simply set `generator_name` we won't have to replace anything as you said

Reply:
Let me know what you think https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2248/commits/5e6ab5c04a6c1dff096d65ff813a2c3a383268cf

Reply:
Good. Can you adjust the PR description as well?

Reply:
Done!